`update_processor_class` has wrong types

**Is this a support request?**

No

**Describe the bug**

When using `Files.new_data_source` as a `update_processor_class` with my Config, as mentioned in the example, mypy throws an error : 

```
error: Argument "update_processor_class" to "Config" has incompatible type "object"; expected "Callable[[Config, FeatureStore, Event], UpdateProcessor] | None"  [arg-type]
```

I believe this is due to the wrong types being specified in either `update_processor_class` or the return type of `new_data_source`.

https://github.com/launchdarkly/python-server-sdk/blob/dc59c94de12d60872475eb3b5b32c03ceac6c7fd/ldclient/integrations/__init__.py#L205-L221

https://github.com/launchdarkly/python-server-sdk/blob/dc59c94de12d60872475eb3b5b32c03ceac6c7fd/ldclient/config.py#L379-L383

**Expected behavior**
There should be no type error when using the method as in the documentation.

**SDK version**
`9.12.0`

--- a/ldclient/integrations/__init__.py
+++ b/ldclient/integrations/__init__.py
@@ -3,8 +3,10 @@ This submodule contains factory/configuration methods for integrating the SDK wi
 other than LaunchDarkly.
 """
 
-from typing import Any, Dict, List, Mapping, Optional
+from threading import Event
+from typing import Any, Callable, Dict, List, Mapping, Optional
 
+from ldclient.config import Config
 from ldclient.feature_store import CacheConfig
 from ldclient.feature_store_helpers import CachingStoreWrapper
 from ldclient.impl.integrations.consul.consul_feature_store import (
@@ -23,7 +25,7 @@ from ldclient.impl.integrations.redis.redis_big_segment_store import (
 from ldclient.impl.integrations.redis.redis_feature_store import (
     _RedisFeatureStoreCore
 )
-from ldclient.interfaces import BigSegmentStore
+from ldclient.interfaces import BigSegmentStore, FeatureStore, UpdateProcessor
 
 
 class Consul:
@@ -206,7 +208,7 @@ class Files:
     """Provides factory methods for integrations with filesystem data."""
 
     @staticmethod
-    def new_data_source(paths: List[str], auto_update: bool = False, poll_interval: float = 1, force_polling: bool = False) -> object:
+    def new_data_source(paths: List[str], auto_update: bool = False, poll_interval: float = 1, force_polling: bool = False) -> Optional[Callable[[Config, FeatureStore, Event], UpdateProcessor]]:
         """Provides a way to use local files as a source of feature flag state. This would typically be
         used in a test environment, to operate using a predetermined feature flag state without an
         actual LaunchDarkly connection.

